website/src/components/SearchPage/fields/LineageField.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState, type FC } from 'react';
+
+import { AutoCompleteField, type Option } from './AutoCompleteField';
+import type { MetadataFilter, SetSomeFieldValues } from '../../../types/config';
+
+interface LineageFieldProps {
+    lapisUrl: string;
+    lapisSearchParameters: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any -- TODO(#3451) use a proper type
+    field: MetadataFilter;
+    fieldValue: string;
+    setSomeFieldValues: SetSomeFieldValues;
+}
+
+export const LineageField: FC<LineageFieldProps> = ({
+    field,
+    fieldValue,
+    setSomeFieldValues,
+    lapisUrl,
+    lapisSearchParameters,
+}) => {
+    const [includeSublineages, setIncludeSubLineages] = useState(fieldValue.endsWith('*'));
+    const [inputText, setInputText] = useState(fieldValue.endsWith('*') ? fieldValue.slice(0, -1) : fieldValue);
+
+    useEffect(() => {
+        let queryText = includeSublineages ? `${inputText}*` : inputText;
+        if (queryText === '*') queryText = '';
+        if (queryText === fieldValue) return;
+        setSomeFieldValues([field.name, queryText]);
+    }, [includeSublineages, inputText, fieldValue]);
+
+    // A function to modify the options available in the dropdown. It adds prefixes.
+    const optionsModifier = (options: Option[]) => {
+        const m = new Map<string, number | undefined>(options.map((o) => [o.option, o.count]));
+        [...m.keys()].forEach((option) => {
+            [...Array(option.length).keys()]
+                .slice(1)
+                .map((i) => option.slice(0, i))
+                .filter((prefix) => !prefix.endsWith('.'))
+                .filter((prefix) => !m.has(prefix))
+                .forEach((prefix) => m.set(prefix, undefined));
```

Comment:
This is specific to Pango lineages (and similar lineage systems) but at least in LAPIS/SILO, it cannot be assumed that the `A` is a parent lineage of `A.1` and that `A` is a valid lineage at all if the data only contain `A.1`.

Reply:
Ah, good to know! But we leave it in for now and can change it later once we also get the lineage definition file from SILO, right?

Reply:
I see.

I have removed that bit of code: https://github.com/loculus-project/loculus/pull/3467/commits/a9d966e801a886820d287db7533410f35bfdb9b7